### PR TITLE
[mission] Adds insertion mode.

### PR DESCRIPTION
**Pull request by Fabien-B**
Adds ReplaceNexts insertion mode

Fixes issue #2014

### CODE REVIEW

**Comment by Fabien-B** [comment]
It allows to replace the next mission element without affecting the current one.

**Comment by podhrmic** [comment]
@Fabien-B can you elaborate please? Also, you can just push the changes into the current pull request (in this case #2014 instead of opening a new one)

**Comment by Fabien-B** [comment]
This insertion mode could be used in a software made for controlling the drone in mission mode (a "dynamic planner" software).
It could be useful to cancel some missions send to the drone but not yet running, without affecting the current behaviour of the drone.

Lets say the drone is currently following a path between 3 waypoints WP1, WP2 and WP3.
The user of the dynamic planner already planned next missions blocks : lets say a circle around WP2 and a GoTo WP4.
While the drone is following the path, the operator change its mind, and wants to replace the next element by an other, for example a survey, or wants to remove the GoTo WP4.

Sending a mission block with REPLACE_ALL insertion mode will leads the drone to an uncertain trajectory, depending on where the drone currently is.

The REPLACE_NEXTS insertion mode could then be used.

File: sw/airborne/modules/mission/mission_common.c [hunk 1]
```diff
@@ -73,6 +73,10 @@ bool mission_insert(enum MissionInsertMode insert, struct _mission_element *elem
       mission.current_idx = 0;
       mission.insert_idx = 1;
       break;
+    case ReplaceNexts:
+        tmp = (mission.current_idx + 1) % MISSION_ELEMENT_NB;
```
Comment (gautierhattenberger) [inline sw/airborne/modules/mission/mission_common.c]: can you follow the 2 spaces indentation

**Comment by Fabien-B** [comment]
Sorry, identation should be ok now.

**Pull request merged**